src/sqlfluff/core/parser/grammar/base.py

```diff
@@ -876,6 +899,8 @@ def copy(
         at: Optional[int] = None,
         before: Optional[Any] = None,
         remove: Optional[List[MatchableType]] = None,
+        terminators: List[Union[str, MatchableType]] = [],
+        add_terminators: List[Union[str, MatchableType]] = [],
```

Comment:
With these situations where we are raising a runtime error if both `terminators` and `add_terminators` are set, I try to look for ways to refactor to ensure that it's impossible instead.

What about one variable for the terminators and one variable about whether to add to existing?
```suggestion
        terminators: List[Union[str, MatchableType]] = [],
        add_to_existing_terminators: bool = False,
```

Or the `terminators` argument itself is a tuple or dataclass or something that has both the list and a boolean or enum expressing whether to add or override.

Reply:
I like that idea. I'll do something like that 👍